### ISSUE TITLE
Enhancement: [prefer-optional-chain] Support simple equals expressions

### ISSUE THREAD

**Issue by boris-petrov** [typescript-eslint/typescript-eslint#7986]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-optional-chain/

### Description

`prefer-optional-chain` could also support cases like the following:

```ts
if (this.foo != null && this.foo.bar === 1) {
}
```

This is probably a very common pattern and would be nice to be handled by the rule.

### Fail

```typescript
if (this.foo != null && this.foo.bar === 1) {
}
```

### Pass

```typescript
if (this.foo?.bar === 1) {
}
```

**Comment by bradzacher** [comment]
This would be covered by https://github.com/typescript-eslint/typescript-eslint/issues/7170
